scripts/build-webpack.js

```diff
@@ -35,6 +35,8 @@ const compiler = webpack({
   output: {
     filename: `yarn-${version}.js`,
     path: path.join(basedir, 'artifacts'),
+    library: 'yarn-cli',
+    libraryTarget: 'commonjs2',
```

Comment:
Yes, this is the solution that fixes the single-file bundles. Without this, Webpack exports an empty object, causing `yarn-bundled-entry-point.js` to fail since `main` function can't be imported.